image: duplicate version caching bugfix

dtorop opened the pull request:
Set duplicate image version in the DB before the cache entry for the image is first allocated. Otherwise, image version in the cache will not reflect the version in the DB.

Also, the change timestamp and group ID for the original image were being written.  Instead, update these for the duplicate.

Fixes #5674.

TurboGit commented on src/common/image.c:
```diff
@@ -935,9 +928,18 @@ int32_t dt_image_duplicate_with_version(const int32_t imgid, const int32_t newve
     g_free(filename);
 
-    const dt_image_t *img = dt_image_cache_get(darktable.image_cache, imgid, 'r');
+    const dt_image_t *img = dt_image_cache_get(darktable.image_cache, newid, 'r');
```
This looks wrong to me. You are setting the group-id to be the one of the new image (the duplicate itself). The group-id of the artifact should be the group of the main image.

dtorop replied:
That makes sense. I'll fix/re-test...

TurboGit replied:
Note that the issue may already be fixed with my latest work on duplicate.

dtorop replied:
I'm just looking at this. This PR fixes something different, that the calls to dt_tag_detach_by_string() and dt_image_cache_unset_change_timestamp() will call dt_image_cache_unset_change_timestamp() which load the version # in from the DB. So it's important to set the version # before it gets cached. (Or an alternative would be to make the version # writeable via the cache.)

Also dt_image_cache_unset_change_timestamp() should be working on the new image, even if I was wrong about the group ID.

dtorop commented:
I fixed https://github.com/darktable-org/darktable/pull/5678#discussion_r448252127 and rebased onto 08c1dcd756079487df79b300893dda7963ebc1f6. This PR should still be necessary to fix #5674, which I can otherwise replicate in current git master.

TurboGit commented:
Ok, works for me. Thanks.
